src/find-all.ts

```diff
@@ -0,0 +1,33 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+
+/**
+ * find all tsconfig.json files in dir
+ *
+ * @param {string} dir - path to dir (absolute or relative to cwd)
+ * @returns {Promise<string[]>} list of absolute paths to all found tsconfig.json files
+ */
+export async function findAll(dir: string): Promise<string[]> {
+	const files = [];
+	for await (const tsconfigFile of findTSConfig(path.resolve(dir))) {
+		files.push(tsconfigFile);
+	}
+	return files;
+}
+
+const ignored_directories = ['node_modules', '.git'];
```

Comment:
Is this correct? technically there can be npm packages including source .ts files and tsconfig.json 
however i don't think it's legal to `import x from 'some-package/src/x.ts'` and expect a local environment to transpile it.
You could also have `/src/node_modules/fake-module/foo.ts` but again, is that something people do?

What other directories could be safely excluded?

some stats:
It takes around 3 seconds for find-all to find all 1015 tsconfig.json files in my home directory without filtering, with filtering node_modules and .git it takes about 1 second for 593 results.

Reply:
I think it is ok.
Maybe it needs to consider about monorepos? (I dont have much experience with monorepos)

Reply:
made it configurable with a skip option